documentation/proposals/Proposal - Multi-Backend Input.md

```diff
@@ -0,0 +1,184 @@
++       /// </summary>
++       void Remove(IInputBackend backend);
++
++       /// <summary>
++       /// Updates all input data on all backends.
++       /// </summary>
++       void Update();
+    }
+}
+```
+
+```diff
+namespace Silk.NET.Input
+{
++   /// <summary>
++   /// Represents an input backend from which input devices can be retrieved.
++   /// </summary>
++   public interface IInputBackend
++   {
++       /// <summary>
++       /// Gets all connected devices recognised by this backend.
++       /// </summary>
++       IReadOnlyList<IInputDevice> ConnectedDevices { get; }
++
++       /// <summary>
++       /// Raised when an input device is connected or disconnected.
++       /// </summary>
++       event Action<IInputDevice, bool>? ConnectionChanged;
++
++       /// <summary>
++       /// Updates this backend's input data.
++       /// </summary>
++       /// <remarks>
++       /// Input events and changes are permitted to occur outside of this method.
++       /// </remarks>
++       void Update();
++
++       /// <summary>
++       /// Activates this input backend and runs any necessary prerequisites.
++       /// </summary>
++       /// <remarks>
++       /// Called by <see cref="InputContext.Add" />
++       /// </remarks>
++       void Activate();
++
++       /// <summary>
++       /// Deactivates this input backend and tears down any resources created.
++       /// </summary>
++       /// <remarks>
++       /// Called by <see cref="InputContext.Remove" />
++       /// </remarks>
++       void Deactivate();
++   }
+}
+```
+
+```diff
+namespace Silk.NET.Input
+{
+-   public static class InputWindowExtensions
+-   {
+-       // ...
+-   }
+}
+```
+
+```diff
+namespace Silk.NET.Input
+{
++   /// <summary>
++   /// Represents the out-of-the-box input backend, which uses a native API to retrieve input backends using a native handle.
++   /// </summary>
++   /// <remarks>
++   /// On desktop, this uses GLFW.
++   /// </remarks>
++   public static class InputBackend
++   {
++       /// <summary>
++       /// Creates an instance of the out-of-the-box input backend, which uses a native API to retrieve input backends using
++       /// the given native handle, optionally calling the given delegates when the input backend is activated and deactivated.
++       /// </summary>
++       /// <param name="handle">The native handle created/sourced from the underlying native API used by this input backend.</param>
++       /// <param name="onActivate">A delegate to be called when the context is activated. May be null.</param>
++       /// <param name="onDeactivate">A delegate to be called when the context is deactivated. May be null.</param>
++       /// <remarks>
++       /// This method is implicitly called by the Windowing-Input integration. <br />
++       /// On desktop, this uses GLFW.
++       /// </remarks>
++       public static unsafe IInputBackend Create(void* handle, Action? onActivate = null, Action? onDeactivate = null);
```

Comment:
This is just very very type-unsafe. How is anyone supposed to know what to pass as `handle` especially considering the only documentation is "On desktop this uses GLFW"?

Reply:
This isn't really intended for general consumption. I'm personally not a fan of changing the public API depending on TFM, but then again this is also not intended for general consumption so I guess I can adjust this.

Reply:
I don't want to change the public API depending on TFM either, but I'd like some documentation on what it does. I think right now you are the only one that knows😅

Reply:
Added a description of what the handle is on GLFW. Other platforms TBD.